typespec-tests/src/test/java/com/type/union/UnionsClientTest.java

```diff
@@ -4,63 +4,117 @@
+    private final MixedTypesClient client10 = new UnionClientBuilder().buildMixedTypesClient();
 
     @Test
-    void sendInt() {
-        client.sendInt(new ModelWithSimpleUnionProperty(BinaryData.fromObject(1)));
+    public void testStringsOnlyClient() {
+        Prop1 prop = client1.get().getProp();
+        Assertions.assertEquals(Prop1.B, prop);
+        client1.send(prop);
     }
 
+    @Disabled("cadl-ranch bug")
     @Test
-    void sendIntArray() {
-        client.sendIntArray(new ModelWithSimpleUnionProperty(BinaryData.fromObject(Arrays.asList(1, 2))));
+    public void testStringExtensibleClient() {
+        Prop prop = client2.get().getProp();
+        Assertions.assertEquals("custom", prop.toString());
+        client2.send(prop);
     }
 
+    @Disabled("cadl-ranch bug")
     @Test
-    void sendFirstNamedUnionValue() {
-        client.sendFirstNamedUnionValue(new ModelWithNamedUnionProperty(BinaryData.fromObject(new Model1("model1", 1))));
+    public void testStringExtensibleNamedClient() {
+        StringExtensibleNamedUnion prop = client3.get().getProp();
+        Assertions.assertEquals("custom", prop.toString());
+        client3.send(prop);
     }
 
     @Test
-    void sendSecondNamedUnionValue() {
-        client.sendSecondNamedUnionValue(new ModelWithNamedUnionProperty(BinaryData.fromObject(new Model2("model2", 2))));
+    public void testIntsOnlyClient() {
+        Prop2 prop = client4.get().getProp();
+        Assertions.assertEquals(2L, prop.toLong());
+        client4.send(prop);
     }
 
     @Test
-    void receiveString() {
-        ModelWithSimpleUnionPropertyInResponse response = client.receiveString();
-        Assertions.assertEquals("string", response.getSimpleUnion().toObject(String.class));
+    public void testFloatsOnlyClient() {
+        Prop3 prop = client5.get().getProp();
+        Assertions.assertEquals(2.2, prop.toDouble());
+        client5.send(prop);
     }
 
     @Test
-    void receiveIntArray() {
-        ModelWithSimpleUnionPropertyInResponse response = client.receiveIntArray();
-        Assertions.assertEquals(Arrays.asList(1, 2), response.getSimpleUnion().toObject(List.class));
+    public void testModelsOnlyClient() {
+        BinaryData prop = client6.get().getProp();
+        Assertions.assertEquals("test", prop.toObject(Cat.class).getName());
+        client6.send(BinaryData.fromObject(new Cat("test")));
     }
 
+    @Disabled("bug on prop.getLr().toObject(Lr.class)")
     @Test
-    void receiveFirstNamedUnionValue() {
-        ModelWithNamedUnionPropertyInResponse response = client.receiveFirstNamedUnionValue();
-        Assertions.assertEquals(1, response.getNamedUnion().toObject(Model1.class).getProp1());
+    public void testEnumsOnlyClient() {
+        EnumsOnlyCases prop = client7.get().getProp();
+        Assertions.assertEquals(Lr.RIGHT, prop.getLr().toObject(Lr.class));
+        Assertions.assertEquals(Ud.UP, prop.getUd().toObject(Ud.class));
```

Comment:
This one maybe a bug of stream-style on `ExpandableStringEnum`.

`toObject(Lr.class)` gives me `null`, instead of `Lr.RIGHT`.